core-web/src/main/java/org/silverpeas/core/web/util/viewgenerator/html/pagination/PaginationSP.java

```diff
@@ -24,200 +24,280 @@
-        result.append("<script type=\"text/javascript\">");
-        result.append("function display").append(jumperName).append("() {");
-        result.append("var ").append(jumperName).append(" = document.getElementById(\"").append(
-            jumperName).append("\");");
-        result.append("if (").append(jumperName).append(".style.visibility != \"visible\") {");
-        result.append("").append(jumperName).append(".style.visibility = \"visible\";");
-        result.append("").append(jumperName).append(".focus();");
-        result.append("} else {");
-        result.append("").append(jumperName).append(".style.visibility = \"hidden\";");
-        result.append("}");
-        result.append("}");
-
-        result.append("function check").append(jumperName).append("Submit(ev) {");
-        result.append("var touche = ev.keyCode;");
-        result.append("if (touche == 13) {");
-        result.append("var index = parseInt(document.getElementById(\"").append(jumperName).append(
-            "\").value);");
-        result.append("if (isNaN(index) || index < 0) { index = 0; }");
-        result.append("if (index > ").append(nbPage).append(") { index = ").append(nbPage).append(
-            "; }");
-        result.append("index = (index-1)*").append(getNbItemsPerPage()).append(";");
-        if (StringUtil.isDefined(javascriptFunc)) {
-          result.append(javascriptFunc).append("(index);");
-        } else {
-          if (getBaseURL() != null) {
-            result.append("location.href=\"").append(getBaseURL()).append("\"+index;");
+  private String getJumperFragment(String beforeScript, String javascriptFunc) {
+    StringBuilder fragment = new StringBuilder();
+    boolean displayJumper = getNbPage() > JUMPER_THRESHOLD;
+    long timeStamp = new Date().getTime();
+    String jumperName = "jumper" + timeStamp;
+    if (displayJumper) {
+      // display page jumper
+      fragment.append("<div class=\"pageJumper\">");
+      fragment.append("<a href=\"javascript:display").append(jumperName).append(
+          "()\" onfocus=\"this.blur()\" title=\"").append(getString("GEF.pagination.jumptoPage"))
+          .append("\">").append(
+          getString("GEF.pagination.jumper")).append(" </a>");
+      fragment
+          .append("<input type=\"text\" class=\"jumper\" id=\"").append(jumperName).append(
+          "\" size=\"3\" onkeydown=\"check").append(jumperName).append("Submit(event)\"/>");
+      fragment.append("</div>\n");
+    }
+
+    fragment.append(beforeScript);
+
+    if (displayJumper) {
+      fragment.append(getJumperScript(jumperName, javascriptFunc));
+    }
+
+    return fragment.toString();
+  }
+
+  private String getNbItemsPerPageFragment(boolean nbItemsPerPage, String javascriptFunc) {
+    StringBuilder fragment = new StringBuilder();
+    if (nbItemsPerPage && getNbItems() > NUMBERPERPAGE_THRESHOLD) {
+      fragment.append("<div class=\"pageIndex numberPerPage\">");
+      int[] values = new int[]{25, 50, 100};
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 25 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 50 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png 'Severity: MINOR') Assign this magic number 100 to a well-named constant, and use the constant instead. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS109)